weave-js/src/components/PagePanelComponents/Home/Browse3/pages/wfReactInterface/tsDataModelHooksCallRefExpansion.ts

```diff
@@ -71,98 +82,350 @@ export const useClientSideCallRefExpansion = (
 const doExpansionIteration = async (
   traceCalls: traceServerClientTypes.TraceCallSchema[],
   expandedRefColumns: Set<string>,
-  client: traceServerClient.TraceServerClient
+  client: traceServerClient.TraceServerClient,
+  recursiveUnwrap: boolean = false,
+  iterationCount: number = 0,
+  maxIterations: number = 10
 ): Promise<traceServerClientTypes.TraceCallSchema[]> => {
+  // Safety check to prevent infinite recursion
+  if (iterationCount >= maxIterations) {
```

Comment:
good safety check, do we also want one one the raw # of traces? one ref could have hundreds of ref children, sort of worried about blowup here